change CI pipeline and website deployment to GitHub Action

Cuda-Chen opened the pull request:
1. Change CI pipeline and website deployment to GitHub Action.
2. Add CI pipeline badge in README.md.

Cuda-Chen committed:
change CI pipeline and website deployment to GitHub Action

1. Change CI pipeline and website deployment to GitHub Action.
2. Add CI pipeline badge in README.md.

Cuda-Chen committed:
remove Travis CI pipeline

c910335 commented on .github/workflows/crystal.yml:
```diff
@@ -0,0 +1,30 @@
+      - name: Install dependencies
+        run: shards install --ignore-crystal-version
+      - name: Run tests
+        run: crystal spec
```
這 repo 沒寫 spec

c910335 commented on .github/workflows/crystal.yml:
```diff
@@ -0,0 +1,30 @@
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: shards install --ignore-crystal-version
```
你可以用 `shards update` 更新一下 `shards.lock`
這裡就不需要後面的參數了

c910335 commented on .github/workflows/website.yml:
```diff
@@ -0,0 +1,26 @@
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Generate website
```
前面少了 `shards build`

c910335 commented on .github/workflows/website.yml:
```diff
@@ -0,0 +1,26 @@
+      - name: Deploy website
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
`personal_token: ${{ secrets.PERSONAL_TOKEN }}`

c910335 commented on .github/workflows/crystal.yml:
```diff
@@ -0,0 +1,30 @@
+      - name: Lint
+        run: bin/ameba
+      - name: Build executable
+        run: shards build --ignore-crystal-version
```
不用參數

Cuda-Chen committed:
apply fixes

c910335 commented on .github/workflows/crystal.yml:
```diff
@@ -0,0 +1,28 @@
+    steps:
+      - uses: actions/checkout@v2
+      - name: Update dependencies
+        run: shards update
```
我是叫你用 `shards update` 更新 `shards.lock`
不是叫你把這行改成 `shards update`

c910335 commented on .github/workflows/crystal.yml:
```diff
@@ -0,0 +1,28 @@
+      - name: Check formatting
+        run: crystal tool format --check
+      - name: Lint
+        run: bin/ameba
```
我沒有說這裡要拿掉 `shards build` 啊

Cuda-Chen committed:
change settings in crystal.yml

1. Change settings in `crystal.yml` for website assets building
test.
2. Update dependencies.

c910335 commented on .github/workflows/website.yml:
```diff
@@ -0,0 +1,28 @@
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build website assets
```
Merge 按下去才發現這行三小

Cuda-Chen replied:
我本來想在 deploy 才叫 `shard build` 把 executable 給 build 出來

c910335 replied:
executable 和 website assets 看起來很像嗎

Cuda-Chen replied:
完全不像
